setup.py

```diff
@@ -0,0 +1,126 @@
+
+with open(path.join(setup_py_location, 'README.md')) as f:
+  readme = f.read()
+
+
+def shogun_cmake(arguments=None):
+    if arguments is None:
+        arguments='-DCMAKE_BUILD_TYPE=Release -DENABLE_TESTING=OFF -DCMAKE_INSTALL_PREFIX=install'
+
+
+    if distutils.spawn.find_executable('cmake') is not None:
+        print('Running CMake...')
+        print('CMake arguments: %s ' % arguments)
+        
+        build_directory = path.join(setup_py_location, 'build')
+        
+        print('Creating build directory: %s' % build_directory)
+        distutils.dir_util.mkpath(build_directory)
+        
+        chdir(build_directory)
+        
+        try:
+            distutils.spawn.spawn(['cmake'] + arguments.split() + ['..'])
+            distutils.spawn.spawn(['make', 'all'])
+            distutils.spawn.spawn(['make', 'install'])
+            
+        except distutils.spawn.DistutilsExecError:
+            print('CMake error.')
+            sys.exit()
+    
+    else:
+        print('CMake is required to build shogun!')
+        sys.exit()
+
+
+# https://docs.python.org/2/distutils/apiref.html#creating-a-new-distutils-command
+class ShogunBuild(distutils.command.build.build):
+    user_options = distutils.command.build.build.user_options + [('cmake=', None, 'Specify CMake arguments.')]
+    build_base = ''
+    build_lib = ''
+    build_scripts = ''
+    plat_name = ''
+
+    def initialize_options(self):
+        self.cmake = None
+    
+    def finalize_options(self):
+        pass
+    
+    def run(self):
+        shogun_cmake(self.cmake)
+    
+    # Command.sub_commands
+
+
+class ShogunInstall(setuptools.command.install.install):
+    def run(self):
+        print('RUN INSTALL')
+        
+        shogun_cmake() 
+        chdir(getcwd())
+        self.do_egg_install() 
+
+shogun_setup(
+    name = 'shogun',
+    version = '4.2.0',
+    description = 'The Shogun Machine Learning Toolbox',
+    long_description = readme,
+    url = 'http://www.shogun-toolbox.org/',
+
+    author = 'Ariane Paola Gomes',
+    author_email = 'arianepaola@users.noreply.github.com',
+
+    license = 'The GNU General Public License v3.0',
+
+    # https://pypi.python.org/pypi?%3Aaction=list_classifiers
+    classifiers = [
+        'Development Status :: 5 - Production/Stable',
+        'Intended Audience :: Developers',
+        'Intended Audience :: Education',
+        'Intended Audience :: Information Technology',
+        'Intended Audience :: Science/Research',
+        'Topic :: Software Development :: Libraries',
+        # Python 2 and Python 3 support
+        'Programming Language :: Python :: 2',
+        'Programming Language :: Python :: 3'
+    ],
+
+    keywords = 'Large scale kernel methods and Support Vector Machines (SVM)',
+    
+    zip_safe = False,
+    
+    cmdclass = {'build': ShogunBuild, 'install': ShogunInstall},
+    
+    packages = ['shogun'],
+
+    install_requires = ['numpy'],
```

Comment:
I know :smile: it is very minimal at the moment, but the dependencies will be added.